src/clj/game/cards/programs.clj

```diff
@@ -435,10 +435,20 @@
    "Imp"
    {:flags {:slow-trash (req (pos? (get-in card [:counter :virus] 0)))}
     :data {:counter {:virus 2}}
-    :abilities [{:counter-cost [:virus 1]
-                 :msg "trash at no cost"
-                 :once :per-turn
-                 :effect (effect (trash-no-cost))}]}
+    :events {:access {:delayed-completion true
+                      :req (req (pos? (get-in card [:counter :virus] 0)))
```

Comment:
The trailing 0/zero in get-in does nothing here.  Not a problem but not needed.

Reply:
The goal is to give a default value if the card returns `nil`. Will it not ever return `nil`?

Reply:
you are right - i retract this one ;)